File: tests/nodejs/cases/case_184_indirect_container_write_09.js
/**
 * Task case 184: indirect_container_write_09 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case184IndirectContainerWrite09(input) {
  const taskName = 'indirect_container_write_09';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const envelope = { wrapped: payload };
  // ESCAPE: payload stored via indirection into retained sink.
  retainedAudit.push(envelope.wrapped);
  return 'ok';
}

module.exports = {
  case184IndirectContainerWrite09,
};

File: tests/go/cases/case_184_indirect_container_write_09.go
package escape_tests

var retainedCase184 = []map[string]string{}

func Case184IndirectContainerWrite09(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "indirect_container_write_09",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "indirect_container_write_09:" + raw,
	}
	envelope := map[string]map[string]string{"wrapped": payload}
	// ESCAPE: indirection writes payload into retained sink.
	retainedCase184 = append(retainedCase184, envelope["wrapped"])
	return "ok"
}
